### PR TITLE
Dev Container config

### PR DESCRIPTION
**Pull request by crazy4pi314**
Adding a container config from the open spec (containers.dev) that will help contributors in VS Code, Codespaces + Jetbrains tools have similar development experiences. It uses a basic Python Docker container, adds a Rust feature, and then installs nox and maturin. I have tested and all of the nox sessions pass.

### CODE REVIEW

**Comment by henryiii** [comment]
Can you allow maintainers edit access?

**Comment by henryiii** [comment]
pre-commit.ci autofix

**Review by henryiii — APPROVED** [review]

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+// README at: https://github.com/devcontainers/templates/tree/main/src/universal
+{
+  "name": "Default Linux Universal",
```
Comment (henryiii) [inline .devcontainer/devcontainer.json:4]:
```suggestion
  "name": "scientific-python-cookie[dev]",
```

Is there anything special with `[dev]`?

Comment (crazy4pi314) [reply .devcontainer/devcontainer.json:4]: Not really, I use that sometimes to indicate that it setups all the dev tooling and not just a user environment.

**Pull request merged**